### ISSUE TITLE
Move extended tests groups into Go packages

### ISSUE THREAD

**Issue by mfojtik** [openshift/origin#4005]
Currently our schema for extended tests groups is following:

```
./default_mysql_test.go
./default_builds_test.go
```

It means that all extended tests are part of the 'extended' Go package. This might get complex when we start adding more extended tests and the package will grow. 
We might want to move those tests into logically Go packages that reflect the groups. So the current schema should be:

```
./images/mysql_test.go
./builds/sti_environment_test.go
```

There are some problems with this approach:

* Ginko doesn't like this pattern as you have to import the package if you want to have one single test runner (and this is hard because _test.go file are not imported by default) or you have to create test runner for each package.
* For some reason, this setup makes the test compilation much slower
* The runner shell script has to be updated (or more smart about selecting groups you want to run).
* Perhaps we can get rid of "build tags" we currently use with this new approach (but that will mean that you must build all tests and then select those you want to run with Ginko)

**Comment by mfojtik** [comment]
@deads2k FYI

**Comment by mfojtik** [comment]
@deads2k I was trying to run the extended tests using `ginkgo -p` which should run all cases in parallel (using N nodes). AFAIK, Ginko compiles every test cases separately and run them in parallel. Problem is that it drops environment variables (KUBECONFIG is not set). Also the `testContext` is empty as we set it in `extended_test.go` but not for every test file... 

I think we should investigate more how to run the cases in parallel (@jhadvig cc)

**Comment by mfojtik** [comment]
@danmcp actually I was planning to work on this tomorrow (if @deads2k has not already started)

**Comment by deads2k** [comment]
> @danmcp actually I was planning to work on this tomorrow (if @deads2k has not already started)

I have not.

**Comment by mfojtik** [comment]
Already done.
